src/main/java/com/codingame/core/GameManager.java

```diff
@@ -0,0 +1,60 @@
+package com.codingame.core;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Singleton;
+
+@Singleton
+public final class GameManager<T extends AbstractPlayer> {
+    private List<T> players;
```

Comment:
What sets players? If it's the GameManager, you need the `class` for `T`.

Reply:
Or maybe you prefer to do a `setPlayers` dans create the players in the main?

Reply:
What about creating 2 Providers, one for the Referee, the other for Players. Then we can inject them. And have a cleaner code.
And Maybe we can create a start() method in the game Manger, like this we just call this start()  from the Main

Reply:
I think, the better way to answer to this kind of question is to provide unit tests. Unit tests will show the API in use…